### ISSUE TITLE
Annotation for balanced structural variants

### ISSUE THREAD

**Issue by dnil** [Ensembl/ensembl-vep#79]
Annotation for balanced structural variants is messy at the moment; e.g. an inversion will show a lot of genes between the breakpoints with all kinds of functional annotations, like start_lost etc as if there was a deletion of all the genes in between. It would be convenient downstream to just have these silenced for balanced events. Ideally of course correctly annotate fusions, rearrangements, reading frame etc, but but leave the start_lost, transcript_ablation etc for the single genes at the very breakpoints? Ofcourse genes around the breaks for an inversion or translocation could be affected - possibly far away - but then typically by changes of expression level. Thanks!

**Comment by willmclaren** [comment]
Agreed, and in general VEP's annotation of SVs is very limited; essentially it only identifies the overlapping features and (in the case of DUPs or DELs) whether they are amplified or deleted. No sequence-level interpretation is carried out, nor is any interpretation of copy number.

It is something we are hoping to improve in a future version.

**Comment by dnil** [comment]
On the same line of issue, only one end of a BND seems to be annotated at the moment. This means a gene will only be annotated for effect if it happens to be in the "ref" coordinate, not if its only in the "alt".

**Comment by henrikstranneheim** [comment]
Any progress on this issue?

**Comment by fgvieira** [comment]
@willmclaren 
Is there an estimate of when such feature will be implemented?
thanks,

**Comment by dnil** [comment]
Right, this is becoming an issue with using VEP. 
Would you welcome a PR on this? 
Perhaps we could pool some resources in getting this together? 
The first level shouldn't be too hard: e.g. fixing inversion annotation. The next step with checking reading frames for tandem dups, dels, fusions and their  potential splice patterns is more involved, but should be quite fun..

**Comment by sarahhunt** [comment]
Hi @dnil , @fgvieira. 
We are planning on improving VEP's annotation of structural variants, but it will probably be next year before major new functionality is available.  We are very interested in specific use cases and example input for problems you are seeing at the moment to help guide development.

Re inversions - I assume  you are currently filtering out transcripts with 'OverlapPC=100.00' where the class is 'INV'? We could flag them as 'transcript_translocation' - would that be more helpful?  

We do welcome PR's and our guidelines are here:
https://github.com/Ensembl/ensembl/blob/release/97/CONTRIBUTING.md

**Comment by yangyxt** [comment]
Found this post in 2022 and wonder whether there are new features available. A bit curious why inversion is not analysed like other basic structural variants. I can convert small INVs to the indel format, but for some predicted big INVs, VEP just doesn't accept that.

**Comment by nuno-agostinho** [comment]
Hi @dnil, @henrikstranneheim, @fgvieira, @yangyxt,

This discussion has been going for quite some years. It would be helpful to have your feedback on the recent VEP updates to better support SVs (please refer to our [changelogs](https://www.ensembl.org/info/docs/tools/vep/script/vep_download.html#new)) and what is still missing.

Thanks for all your help!

Best,
Nuno

**Comment by nuno-agostinho** [comment]
Hi all,

I'll close this issue for now. Hopefully, the new structural variant features in VEP meet your needs. If you encounter any further issues, please don't hesitate to open a new issue. Thanks!

Kind regards,
Nuno

**Comment by dnil** [comment]
Hi @nuno-agostinho! Sorry, missed this. The new SV annotations are indeed very helpful for BNDs and in particular for actually giving information on both ends. Thank you! We were super, super happy - both for the specifics, and in particular to see the field moving this direction again.

I believe we are still having issues with consequence annotation of the INVs, with imprecise interpretation of the consequences of breakpoints for DUPs and to a lesser extent with the annotation of consequences for breakpoints of DELs. But, let us check that more precisely and each can open more detailed issues regarding them separately instead of carrying around a project level issue!

**Comment by nuno-agostinho** [comment]
Hi @dnil,

Thanks for your reply! It's great to hear that you're pleased with the latest VEP updates.

Please do open new issues with detailed feedback on structural variants that you think that could improve VEP.

Have a great day!

Cheers,
Nuno
